src/index.ts

```diff
@@ -1,50 +1,40 @@
-import Eth from 'ethjs';
-import registryMap from './registry-map.json';
-import abi from './abi.json';
+import type { Provider } from '@ethersproject/abstract-provider';
+import { Contract } from '@ethersproject/contracts';
 
-interface HttpProvider {
-  host: string;
-  timeout: number;
-}
+import abi from './abi.json';
+import registryMap from './registry-map.json';
 
 interface MethodRegistryArgs {
   network: string;
-  provider: HttpProvider;
-}
-
-interface DeployedRegistryContract {
-  entries (bytes: string): string[];
+  provider: Provider;
```

Comment:
It might be better to expect an EIP-1193 provider here rather than an ethers provider.